src/main/java/org/folio/orders/utils/StreamUtils.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.orders.utils;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class allowing to perform stream operations in a more convenient way
+ */
+@UtilityClass
+public class StreamUtils {
+
+  public static <T> T find(Collection<T> collection, Predicate<T> predicate) {
+    for (T t : collection) {
+      if (predicate.test(t)) {
+        return t;
+      }
+    }
+    return null;
+  }
+
+  public static <T> List<T> filter(Collection<T> collection, Predicate<T> predicate) {
+    return collection.stream().filter(predicate).toList();
+  }
+
+  public static <K, V> List<V> map(Collection<K> collection, Function<K, V> mapper) {
+    return collection.stream().map(mapper).toList();
+  }
+
+  public static <K, V> Set<V> mapToSet(Collection<K> collection, Function<K, V> mapper) {
+    return collection.stream().map(mapper).collect(Collectors.toSet());
+  }
+
+  public static <T, K> Map<K, T> list2map(Collection<T> collection, Function<T, K> toKey) {
```

Comment:
if we have already in contract of this class mapToSet, so better to name this method as listToMap